From 3611dc18bae543e02dd1313b6330d77165a87fe5 Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Sun, 18 Jun 2023 14:30:14 +0800
Subject: [PATCH 101/125] va: Only change video format in
 gst_va_video_info_from_dma_info()

The current way of using gst_video_info_set_format() will change all
fields of the GstVideoInfo. We only need to change its format, stride
and offset fields.
---
 .../gst-libs/gst/va/gstvavideoformat.c          | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/subprojects/gst-plugins-bad/gst-libs/gst/va/gstvavideoformat.c b/subprojects/gst-plugins-bad/gst-libs/gst/va/gstvavideoformat.c
index e173501ed1..2915ff6633 100644
--- a/subprojects/gst-plugins-bad/gst-libs/gst/va/gstvavideoformat.c
+++ b/subprojects/gst-plugins-bad/gst-libs/gst/va/gstvavideoformat.c
@@ -369,6 +369,8 @@ gst_va_video_info_from_dma_info (GstVideoInfo * info,
     const GstVideoInfoDmaDrm * drm_info)
 {
   GstVideoFormat video_format;
+  GstVideoInfo tmp_info;
+  guint i;
 
   g_return_val_if_fail (drm_info, FALSE);
   g_return_val_if_fail (info, FALSE);
@@ -387,12 +389,19 @@ gst_va_video_info_from_dma_info (GstVideoInfo * info,
   if (video_format == GST_VIDEO_FORMAT_UNKNOWN)
     return FALSE;
 
-  *info = drm_info->vinfo;
-
-  if (!gst_video_info_set_format (info, video_format,
-          GST_VIDEO_INFO_WIDTH (info), GST_VIDEO_INFO_HEIGHT (info)))
+  if (!gst_video_info_set_format (&tmp_info, video_format,
+          GST_VIDEO_INFO_WIDTH (&drm_info->vinfo),
+          GST_VIDEO_INFO_HEIGHT (&drm_info->vinfo)))
     return FALSE;
 
+  *info = drm_info->vinfo;
+  info->finfo = tmp_info.finfo;
+  for (i = 0; i < GST_VIDEO_MAX_PLANES; i++)
+    info->stride[i] = tmp_info.stride[i];
+  for (i = 0; i < GST_VIDEO_MAX_PLANES; i++)
+    info->offset[i] = tmp_info.offset[i];
+  info->size = tmp_info.size;
+
   return TRUE;
 }
 
-- 
2.25.1

